File: sdk/geofence/iOS/MoEReactGeofence.m
#import "MoEReactGeofence.h"

@implementation MoEReactGeofence

RCT_EXPORT_MODULE()
RCT_EXPORT_METHOD(startGeofenceMonitoring:(NSDictionary *) payload) {
    [[MoEngagePluginGeofenceBridge sharedInstance] startGeofenceMonitoring: payload];
}

RCT_EXPORT_METHOD(stopGeofenceMonitoring:(NSDictionary *) payload) {
    [[MoEngagePluginGeofenceBridge sharedInstance] stopGeofenceMonitoring:payload];
}

@end

File: sdk/geofence/android/src/newarch/java/com/moengage/react/geofence/MoEReactGeofence.kt
package com.moengage.react.geofence

import com.facebook.react.bridge.ReactApplicationContext

/**
 * Bridge to communicate with React-Native Goefence Plugin in old arch
 *
 * @author Abhishek Kumar
 * @since Todo: Add Version
 */
class MoEReactGeofence(reactContext: ReactApplicationContext) :
    NativeMoEngageGeofenceSpec(reactContext) {

    private val bridgeHandler = MoEngageGeofenceHandler(reactContext)

    override fun getName() = bridgeHandler.getName()

    override fun startGeofenceMonitoring(payload: String) {
        bridgeHandler.startGeofenceMonitoring(payload)
    }

    override fun stopGeofenceMonitoring(payload: String) {
        bridgeHandler.stopGeofenceMonitoring(payload)
    }
}